charmcraft/services/package.py

```diff
@@ -122,19 +122,23 @@ def pack_charm(
     def get_charm_path(self, dest_dir: pathlib.Path) -> pathlib.Path:
         """Get a charm file name for the appropriate set of run-on bases."""
         if self._platform:
-            return dest_dir / f"{self._project.name}_{self._platform}.charm"
+            platform = self._platform.replace(":", "-")
+            return dest_dir / f"{self._project.name}_{platform}.charm"
         build_plan = models.CharmcraftBuildPlanner.model_validate(
             self._project.marshal()
         ).get_build_plan()
-        platform = utils.get_os_platform()
-        build_on_base = bases.BaseName(name=platform.system, version=platform.release)
+        host_platform = utils.get_os_platform()
+        build_on_base = bases.BaseName(
+            name=host_platform.system, version=host_platform.release
+        )
         host_arch = util.get_host_architecture()
         for build_info in build_plan:
             print(build_info)
             if build_info.build_on != host_arch:
                 continue
             if build_info.base == build_on_base:
-                return dest_dir / f"{self._project.name}_{build_info.platform}.charm"
+                platform = build_info.platform.replace(":", "-")
+                return dest_dir / f"{self._project.name}_{platform}.charm"
 
         raise errors.CraftError(
             "Current machine is not a valid build platform for this charm.",
```

Comment:
Unless I am mistaken, we could delete all of this.  

I don't think there is a scenario where `self._platform` doesn't exist.

Reply:
This may well be true now. It wasn't true when I first implemented this service, but at that point we could have `self._platform == None`. I'd have to do a deep dive into craft-application to see if that's still valid in the PackageService.

Reply:
Makes sense, I was suspecting this was needed during the craft-application migration.

I poked around at the service factory and how charmcraft inits the pack service and dropping this appears safe.